source/language/classical.rst

```diff
@@ -95,11 +116,59 @@ Integer types support addition ``+``, subtraction ``-``, multiplication ``*``, i
    a ** b; // 8
    a += 4; // a == 6
 
-Floating-point numbers and angles
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Floating-point and angle types support addition, subtraction,
-multiplication, division, and power and the corresponding assignment operators.
+Angles
+~~~~~~
+
+In addition to the bitwise operations mentioned above, angles support:
+
+- Addition ``+`` and subtraction ``-`` by other angles of the same size, which
+  returns an angle of the same size.
+- Multiplication ``*`` and division ``/`` by unsigned integers of the same size.
+  The result is an ``angle`` type of the same size.  Both ``uint * angle`` and
+  ``angle * uint`` are valid and produce the same result, but only ``angle /
+  uint`` is valid; it is not allowed to divide an integer by an angle.
+- Division ``/`` by another angle of the same size.  This returns a ``uint`` of
+  the same size.
+- Unary negation ``-``, which represents the mathematical operation :math:`-a
+  \equiv 2\pi - a`.
+- Compound assignment operators ``+=``, ``-=`` and ``/=`` with angles of the
+  same size as both left- and right operands.  These have the same effect as if
+  the equivalent binary operation had been written out in full.
+- The compound assignment operators ``*=`` and ``/=`` with an unsigned integer
+  of the same size as the right operand.  This has the same effect as if the
+  multiplication or division had been written as a binary operation and
+  assigned.
+
+In all of these cases, except for unary negation, the bit pattern of the result
+of these operations is the same as if the operations had been carried out
+between two ``uint`` types of the same size with the same bit representations,
+including both upper and lower overflow.  Explicitly::
+
+  angle[4] a = 7 * (pi / 8);  // "0111"
+  angle[4] b = pi / 8;        // "0001"
+  angle[4] c = 5 * (pi / 4);  // "1010"
+  uint[4] two = 2;
+
+  a + b;    // angle[4] │ pi           │ "1000"
+  b - a;    // angle[4] │ 5 * (pi / 4) │ "1010"
+  a / two;  // angle[4] │ 3 * (pi / 8) │ "0011"
+  two * c;  // angle[4] │ pi / 2       │ "0100"
+  c / b;    // uint[4]  │ 10           │ "1010"
+  pi * 2;   // angle[4] │ 0            │ "0000"
+
+Unary negation of an angle ``a`` is defined to produce the same value as
+``0 - a``, such that ``a + (-a)`` is always equal to zero.  This is the same as
+the C99 definition for unsigned integers.  In bitwise operations, the negation
+can be written as ``(~a) + 1``. Explicitly::
+
+  angle[4] a = pi / 4;  // "0010"
+  angle[4] b = -a;  // 7*(pi/4) │ "1110"
+
+Floating-point numbers
+~~~~~~~~~~~~~~~~~~~~~~
+
+Floating-point numbers support addition, subtraction, multiplication, division,
+and power and the corresponding assignment operators.
```

Comment:
is modulo intentionally excluded here?

Reply:
I didn't change this text, except the first couple of words used to be "floating-point numbers and angles support ...".  I guess whoever originally wrote this did it thinking more about the C spec, where the `%` operator only works on integers.  We have the `mod` function, which does work on floating-point numbers, though.